src/redteam/extraction/purpose.ts

```diff
@@ -0,0 +1,26 @@
+import dedent from 'dedent';
+import { ApiProvider } from '../../types';
+import { callExtraction, formatPrompts } from './util';
+
+export async function extractSystemPurpose(
+  provider: ApiProvider,
+  prompts: string[],
+): Promise<string> {
+  const prompt = dedent`
+    The following are prompts that are being used to test an LLM application:
+    
+    ${formatPrompts(prompts)}
+    
+    Given the above prompts, output the "system purpose" of the application in a single sentence, enclosed in <purpose> tags.
+    
+    Example outputs:
+    <purpose>Provide users a way to manage finances</purpose>
```

Comment:
@typpo it feels sloppy to mix `xml` with `key:`. Should we standardize on one and if so, which one?

Reply:
Given that this is a single output, do we need it to wrap the output in anything?

Reply:
Fair question! We don't need to do this but we should do something more than what we are currently doing. I noticed that sometimes the purpose string sometimes begins with ` -` eg ` - Executive assistant that helps with scheduling and reminders` is the purpose, not `Executive assistant that helps with scheduling and reminders`. I haven't spent time evaluating whether or not this matters at all.